src/sys.c

```diff
@@ -175,6 +176,7 @@ int check_loaded_modules(void)
 		}
 		if (list[n] != NULL) free(list[n]);
 	}
+	free(list);
```

Comment:
So, there is a bigger "leak" here and while my `if (list) free(list)` comment still applies here, like above. Notice that we will exit on error at line 168/169 (the `return rc`) without freeing the array of structures because below, I am using them still.

Here is the thing though, it returns and exits the application, thereby freeing the memory regardless.